### PR TITLE
Rework epc2 build tools

### PR DESCRIPTION
**Pull request by hhoegelo**

### CODE REVIEW

**Comment by AntonSchmied** [comment]
While trying "make epc2-create-iso errors occurred (in case it helps):
````
[ 75%] Create Docker container for epc2os rootfs creation
Sending build context to Docker daemon  17.92kB
Step 1/6 : FROM archlinux:20200908
 ---> 90aeadc1688f
Step 2/6 : COPY createRootFS.sh /
 ---> Using cache
 ---> 75ce8a480402
Step 3/6 : RUN pacman --noconfirm -Syyuu
 ---> Running in 2d8e64a41f53
:: Synchronizing package databases...
downloading core.db...
downloading extra.db...
downloading community.db...
:: Starting full system upgrade...
:: Replace libutil-linux with core/util-linux-libs? [Y/n] 
resolving dependencies...
looking for conflicting packages...
...
...
checking keyring...
checking package integrity...
:: File /var/cache/pacman/pkg/bzip2-1.0.8-4-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: bzip2: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust
error: 
pam: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust
:: File /var/cache/pacman/pkg/pam-1.5.2-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: libunistring: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust

:: File /var/cache/pacman/pkg/libunistring-0.9.10-3-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] 
:: File /var/cache/pacman/pkg/pinentry-1.2.0-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] 
:: File /var/cache/pacman/pkg/perl-5.34.0-3-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: pinentry: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust
error: perl: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust
error: procps-ng: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust

:: File /var/cache/pacman/pkg/procps-ng-3.3.17-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: failed to commit transaction (invalid or corrupted package)

Errors occurred, no packages were upgraded.
The command '/bin/sh -c pacman --noconfirm -Syyuu' returned a non-zero code: 1
make[3]: *** [build-tools/epc2/build-rootfs/CMakeFiles/epc2-build-rootfs.dir/build.make:70: build-tools/epc2/build-rootfs/.build-rootfs-docker-container] Error 1
make[2]: *** [CMakeFiles/Makefile2:1409: build-tools/epc2/build-rootfs/CMakeFiles/epc2-build-rootfs.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1443: build-tools/epc2/create-install-medium/CMakeFiles/epc2-create-iso.dir/rule] Error 2
make: *** [Makefile:502: epc2-create-iso] Error 2
```

**Comment by AntonSchmied** [comment]
and if starting from scratch, these errors occur:
```
[  8%] Build 'Collect Update Packages'-Docker Container
Sending build context to Docker daemon  4.608kB
Step 1/15 : FROM archlinux:20200908
 ---> 90aeadc1688f
Step 2/15 : RUN echo "Server = https://archive.archlinux.org/repos/2020/09/08/\$repo/os/\$arch" > /etc/pacman.d/mirrorlist
 ---> Using cache
 ---> 7f6855b32085
Step 3/15 : RUN mkdir /packages
 ---> Using cache
 ---> ae175020a7b8
Step 4/15 : COPY linux-rt-5.9.1.20.arch1-1-x86_64.pkg.tar.zst /packages/linux-rt-5.9.1.20.arch1-1-x86_64.pkg.tar.zst
COPY failed: file not found in build context or excluded by .dockerignore: stat linux-rt-5.9.1.20.arch1-1-x86_64.pkg.tar.zst: file does not exist
make[3]: *** [build-tools/epc2/collect-packages/CMakeFiles/epc2-collect-packages.dir/build.make:92: build-tools/epc2/collect-packages/.collect-update-packages-docker-container] Error 1
make[2]: *** [CMakeFiles/Makefile2:1272: build-tools/epc2/collect-packages/CMakeFiles/epc2-collect-packages.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1443: build-tools/epc2/create-install-medium/CMakeFiles/epc2-create-iso.dir/rule] Error 2
make: *** [Makefile:502: epc2-create-iso] Error 2
```

**Comment by hhoegelo** [comment]
> and if starting from scratch, these errors occur:
> 
> ```
> [  8%] Build 'Collect Update Packages'-Docker Container
> Sending build context to Docker daemon  4.608kB
> Step 1/15 : FROM archlinux:20200908
>  ---> 90aeadc1688f
> Step 2/15 : RUN echo "Server = https://archive.archlinux.org/repos/2020/09/08/\$repo/os/\$arch" > /etc/pacman.d/mirrorlist
>  ---> Using cache
>  ---> 7f6855b32085
> Step 3/15 : RUN mkdir /packages
>  ---> Using cache
>  ---> ae175020a7b8
> Step 4/15 : COPY linux-rt-5.9.1.20.arch1-1-x86_64.pkg.tar.zst /packages/linux-rt-5.9.1.20.arch1-1-x86_64.pkg.tar.zst
> COPY failed: file not found in build context or excluded by .dockerignore: stat linux-rt-5.9.1.20.arch1-1-x86_64.pkg.tar.zst: file does not exist
> make[3]: *** [build-tools/epc2/collect-packages/CMakeFiles/epc2-collect-packages.dir/build.make:92: build-tools/epc2/collect-packages/.collect-update-packages-docker-container] Error 1
> make[2]: *** [CMakeFiles/Makefile2:1272: build-tools/epc2/collect-packages/CMakeFiles/epc2-collect-packages.dir/all] Error 2
> make[1]: *** [CMakeFiles/Makefile2:1443: build-tools/epc2/create-install-medium/CMakeFiles/epc2-create-iso.dir/rule] Error 2
> make: *** [Makefile:502: epc2-create-iso] Error 2
> ```

is gefixed

**Comment by AntonSchmied** [comment]
I afraid i don't get passed these errors now, when building epc2 from scratch
```
[ 75%] Create Docker container for epc2os rootfs creation
Sending build context to Docker daemon  17.92kB
Step 1/6 : FROM archlinux:20200908
 ---> 90aeadc1688f
Step 2/6 : COPY createRootFS.sh /
 ---> Using cache
 ---> 75ce8a480402
Step 3/6 : RUN pacman --noconfirm -Syyuu
 ---> Running in 719ec98cf341
:: Synchronizing package databases...
downloading core.db...
downloading extra.db...
downloading community.db...
:: Starting full system upgrade...
:: Replace hwids with core/hwdata? [Y/n] 
:: Replace libutil-linux with core/util-linux-libs? [Y/n] 
resolving dependencies...
looking for conflicting packages..
[...]
checking keyring...
checking package integrity...
:: File /var/cache/pacman/pkg/bzip2-1.0.8-4-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: bzip2: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust

:: File /var/cache/pacman/pkg/pam-1.5.2-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: pam: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust
error: libunistring: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust

:: File /var/cache/pacman/pkg/libunistring-0.9.10-3-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: pinentry: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust

:: File /var/cache/pacman/pkg/pinentry-1.2.0-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: perl: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust

:: File /var/cache/pacman/pkg/perl-5.34.0-3-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: procps-ng: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust

:: File /var/cache/pacman/pkg/procps-ng-3.3.17-1-x86_64.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n] error: failed to commit transaction (invalid or corrupted package)

Errors occurred, no packages were upgraded.
The command '/bin/sh -c pacman --noconfirm -Syyuu' returned a non-zero code: 1
make[3]: *** [build-tools/epc2/build-rootfs/CMakeFiles/epc2-build-rootfs.dir/build.make:70: build-tools/epc2/build-rootfs/.build-rootfs-docker-container] Error 1
make[2]: *** [CMakeFiles/Makefile2:1409: build-tools/epc2/build-rootfs/CMakeFiles/epc2-build-rootfs.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1443: build-tools/epc2/create-install-medium/CMakeFiles/epc2-create-iso.dir/rule] Error 2
make: *** [Makefile:502: epc2-create-iso] Error 2
```

File: build-tools/epc2/create-install-medium/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM archlinux:20200908
+
+COPY createIso.sh /
+
+RUN pacman --noconfirm -Syyuu
+RUN pacman --noconfirm -Syy
+
```
Comment (AntonSchmied) [inline build-tools/epc2/create-install-medium/Dockerfile:7]: In order to be able to run epc2-create-iso successfully, lines 5-6 had to be changed to
```
RUN pacman --noconfirm -Sy
RUN pacman --noconfirm -S archlinux-keyring
RUN pacman --noconfirm -Syyuu
RUN pacman --noconfirm -Syy
RUN pacman --noconfirm -S archlinux-keyring
RUN pacman --noconfirm -S arch-install-scripts pacman-contrib dosfstools
```
Not sure if `RUN pacman --noconfirm -S arch-install-scripts pacman-contrib dosfstools` is really needed here

File: build-tools/epc2/build-rootfs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Build environment for arch rootfs creation
+
+FROM archlinux:20200908
+
+COPY createRootFS.sh /
+
+RUN pacman --noconfirm -Syyuu 
+RUN pacman --noconfirm -Syy 
+RUN pacman --noconfirm -S arch-install-scripts pacman-contrib dosfstools
+
```
Comment (AntonSchmied) [inline build-tools/epc2/build-rootfs/Dockerfile:10]: In order to be able to run epc2-create-iso successfully, lines 7-9 had to be changed to
```
RUN pacman --noconfirm -Sy
RUN pacman --noconfirm -S archlinux-keyring
RUN pacman --noconfirm -Syyuu
RUN pacman --noconfirm -Syy
RUN pacman --noconfirm -S archlinux-keyring
RUN pacman --noconfirm -S arch-install-scripts pacman-contrib dosfstools
```

**Comment by AntonSchmied** [comment]
closes #2947

**Pull request merged**